sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -844,8 +844,13 @@ case class ShowTablesCommand(
         val tableName = tableIdent.table
         val isTemp = catalog.isTempView(tableIdent)
         if (isExtended) {
-          val information = catalog.getTempViewOrPermanentTableMetadata(tableIdent).simpleString
-          Row(database, tableName, isTemp, s"$information\n")
+          val information = catalog.getTempViewOrPermanentTableMetadata(tableIdent)
+          val value = if (conf.getConf(SQLConf.LEGACY_KEEP_COMMAND_OUTPUT_SCHEMA)) {
+            s"$information\n"
+          } else {
+            information.toLinkedHashMap
```

Comment:
The each map value, I think it's better to be a single-line string. Shall we update how we turn the schema into string? Currently the schema string is multi-lines

Reply:
+1.

Reply:
Before:
```
root
 |-- a: integer (nullable = true)
 |-- b: string (nullable = true)
 |-- c: integer (nullable = true)

```
After:
```
struct<a:int,b:string,c:int>
```